x-pack/plugins/alerting/server/application/maintenance_window/methods/get_active/get_active_maintenance_windows.ts

```diff
@@ -23,15 +23,29 @@ export interface MaintenanceWindowAggregationResult {
 }
 
 export async function getActiveMaintenanceWindows(
-  context: MaintenanceWindowClientContext
+  context: MaintenanceWindowClientContext,
+  cacheIntervalMs?: number
 ): Promise<MaintenanceWindow[]> {
   const { savedObjectsClient, logger } = context;
 
   const startDate = new Date();
   const startDateISO = startDate.toISOString();
 
+  let eventsKuery: KueryNode;
+  if (cacheIntervalMs) {
+    // add offset to startDate
+    const startDateWithCacheOffset = new Date(startDate.getTime() + cacheIntervalMs);
+    const startDateWithCacheOffsetISO = startDateWithCacheOffset.toISOString();
+    eventsKuery = nodeBuilder.or([
+      nodeBuilder.is('maintenance-window.attributes.events', startDateISO),
```

Comment:
I was confused as well but I tested it and it does work. I think because `events` is mapped as a `date_range` it just works? cc @JiaweiWu do you know?